src/main/scala/io/iohk/ethereum/network/PeerActor.scala

```diff
@@ -177,38 +181,36 @@ class PeerActor(
     case GetStatus => sender() ! StatusResponse(Handshaking)
   }
 
-  def waitingForChainForkCheck(rlpxConnection: RLPxConnection, remoteStatus: msg.Status, timeout: Cancellable): Receive =
+  def waitingForForkHeader(rlpxConnection: RLPxConnection, remoteStatus: msg.Status, timeout: Cancellable,
+                           forkResolver: ForkResolver): Receive =
     handleSubscriptions orElse handleTerminated(rlpxConnection) orElse
     handleDisconnectMsg orElse handlePingMsg(rlpxConnection) orElse handlePeerChainCheck(rlpxConnection) orElse {
 
     case RLPxConnectionHandler.MessageReceived(msg @ BlockHeaders(blockHeaders)) =>
       timeout.cancel()
 
-      val daoBlockHeaderOpt = blockHeaders.find(_.number == daoForkBlockNumber)
-
-      daoBlockHeaderOpt match {
-        case Some(_) if daoForkValidator.validate(msg).isEmpty =>
-          log.info("Peer is running the ETC chain")
-          context become new MessageHandler(rlpxConnection, remoteStatus, daoForkBlockNumber, Chain.ETC).receive
+      val forkBlockHeaderOpt = blockHeaders.find(_.number == forkResolver.forkBlockNumber)
 
-        case Some(_) if getBestBlockHeader().difficulty < daoForkBlockTotalDifficulty =>
-          log.warning("Peer is not running the ETC fork, but we're not there yet. Keeping the connection until then.")
-          context become new MessageHandler(rlpxConnection, remoteStatus, daoForkBlockNumber, Chain.ETH).receive
+      forkBlockHeaderOpt match {
+        case Some(forkBlockHeader) if forkBlockHeader.number == forkResolver.forkBlockNumber =>
```

Comment:
Minor detail: How is the case when `forkBlockHeader.number != forkResolver.forkBlockNumber` is handled? Wouldn't it always be true?